crates/red_knot/src/symbols.rs

```diff
@@ -67,7 +67,10 @@ pub(crate) enum ScopeKind {
 pub(crate) struct Scope {
     name: Name,
     kind: ScopeKind,
-    child_scopes: Vec<ScopeId>,
+    parent: Option<ScopeId>,
+    children: Vec<ScopeId>,
+    definition: Option<Definition>,
+    defining_symbol: Option<SymbolId>,
```

Comment:
Yes I can add docs for these.

We cannot navigate from `SymbolId` to `Definition` because a symbol can have multiple definitions, and we need to know which specific definition. This is a general problem that affected a lot of things in the design of this PR.

Reply:
Yeah, that might make sense. I'll add a TODO and leave that for another PR though.